packages/bundler/src/es6-rewriter.ts

```diff
@@ -106,20 +120,37 @@ export class Es6Rewriter {
                   }
                 }
               }
-              return document.parsedDocument.contents;
+
+              // If the URL of the requested document is the same as the bundle
+              // URL or the requested file doesn't use `import.meta` anywhere,
+              // we can return it as-is.
+              if (this.bundle.url === id ||
+                  !document.parsedDocument.contents.includes('import.meta')) {
+                return document.parsedDocument.contents;
+              }
+
+              // We need to rewrite instances of `import.meta` in the document
+              // to preserve its location because `import.meta` is used and the
+              // URL has changed as a result of bundling.
+              const relativeUrl =
+                  ensureLeadingDot(this.bundler.analyzer.urlResolver.relative(
+                      this.bundle.url, id as ResolvedUrl));
```

Comment:
Is the cast of `id` needed? It looks like it's a ResolvedUrl already

Reply:
You're right. Fixed.